headers/xdp/xsk.h

```diff
@@ -234,6 +260,12 @@ int xsk_umem__create_with_fd(struct xsk_umem **umem,
 			     struct xsk_ring_prod *fill,
 			     struct xsk_ring_cons *comp,
 			     const struct xsk_umem_config *config);
+/* Newer version to create umem by opts, recommended to use. */
```

Comment:
Would we even like to say that the old interface is to be deprecated?

Reply:
Yeah, but let's merge as-is and then think about how we are going to mark things for deprecation as a follow-up :)

Reply:
Let me know when you plan for a 2.0 version. At that point we should convert all the xsk calls to this new structure.

Reply:
And we can use a new API `xsk_socket__create_opts()` to replace the existing `xsk_socket__create()` and `xsk_socket__create_shared()` by adding another opts struct like `xsk_socket_opts`, WDYT? @tohojo @magnus-karlsson

Reply:
Sounds like a good idea!